Tribler/community/tunnel/tunnel_community.py

```diff
@@ -174,9 +192,12 @@ def tunnel_data(self, source, data):
 
     def close(self):
         """
-        Closes the UDP socket if enabled.
+        Closes the UDP socket if enabled and cancels all pending defererds,
         :return: A deferred that fires once the UDP socket has closed.
         """
+        for deferred in self.deferred_list:
+            deferred.cancel()
```

Comment:
If I remember correctly, the deferred returned by `reactor.resolve()` doesn't have a canceler.